racetrack_client/racetrack_client/client/deploy.py

```diff
@@ -164,6 +164,13 @@ def _wait_for_deployment_result(lifecycle_url: str, deploy_id: str, user_auth: s
             logger.info(f'deployment in progress: {phase}...')
         else:
             logger.info(f'deployment in progress...')
+
+    warnings = response['warnings']
+    if not warningss or warningss[-1] != warnings:  # don't print the same warnings again
+        warningss.append(warnings)
+        if warnings:
+            logger.warning(warnings)
```

Comment:
I think it should be moved above the lines
```python
status = response['status'].lower()
    if status == 'failed':
        raise RuntimeError(response['error'])
    if status == 'done':
        return response.get('job', {}).get('pub_url', '')
```
so that we don't miss a warning if it's been applied at the very end of deployment. Note that we don't care about phases, so that's the only difference between phases and warnings.

Reply:
good call